src/app/shared/components/request-card-details/request-card-details.component.ts

```diff
@@ -32,48 +32,59 @@ import { NotificationService } from '../notification/notification.service';
 import * as dayjs from 'dayjs';
 import { group } from '@angular/animations';
 import { MatMenuTrigger } from '@angular/material/menu';
+import { S3Service } from 'src/app/core/services/s3.service';
 
 @Component({
   selector: 'app-request-card-details',
   templateUrl: './request-card-details.component.html',
   styleUrls: ['./request-card-details.component.scss'],
 })
 export class RequestCardDetailsComponent implements OnInit {
-  
   @ViewChild('menuTrigger') menuTrigger: MatMenuTrigger | undefined;
   supporters: any[] = [];
-  lastestComment: { content: string; postTime: string; }[] | undefined;
+  lastestComment: { content: string; postTime: string }[] | undefined;
 
   new_status: String = '';
   cur_status?: String = this.request.status;
   isOpen: boolean = false;
   status: string[] = ['verified', 'accepted', 'rejected'];
-  mapStatus!: Map<string, IBaseStatus>; 
+  mapStatus!: Map<string, IBaseStatus>;
   mapSupportStatus!: Map<string, IBaseStatus>;
-  mapPriority: any
+  mapPriority: any;
   news: INew[] = [];
   user: any;
-  create_time:string='';
+  create_time: string = '';
   trans: ITransaction[] = [];
   supportObject: ISupport[] = [];
   defaultComment: INew = {
     subject: 'new_comment',
     content: '',
+    medias: [],
     target_type: 'sos_request',
     target_id: this.request.id,
   };
+  preUploadFile: any;
+  file: any;
   onClose() {
     this.bottomRef.dismiss(this.request);
   }
   mark($event: any, action?: string) {
     console.log(action);
     $event.stopPropagation();
     $event.preventDefault();
-    this.UrgentRequestService.markRequest(this.request?.id,
-      { bookmarker_type: 'user', action: action, bookmarker_id: this.user.id })
-      .subscribe((res) => {
-        if (action == 'bookmark') { console.log(true); this.request!.is_bookmarked = true; } else { console.log("else"); this.request!.is_bookmarked = false; }
-      })
+    this.UrgentRequestService.markRequest(this.request?.id, {
+      bookmarker_type: 'user',
+      action: action,
+      bookmarker_id: this.user.id,
+    }).subscribe((res) => {
+      if (action == 'bookmark') {
+        console.log(true);
```

Comment:
nguyeen cais nay em nen de: `this.bookmark = action === 'bookmark'`

Reply:
không có biến bookmark được khởi tạo từ đầu nên em dùng === nha anh